File: PrimeraPregunta.hs

calculadora :: Fractional a => a -> a -> Char -> a
calculadora num1 num2 op
    | op == '*'   = multiplicacion num1 num2
    | op == '+'   = suma num1 num2
    | op == '-'   = resta num1 num2
    | op == '/'   = divide num1 num2

suma :: Num a => a -> a -> a
suma num1 num2 = num1 + num2
resta :: Num a => a -> a -> a
resta num1 num2 = num1 - num2
multiplicacion :: Num a => a -> a -> a
multiplicacion num1 num2 = num1 * num2
divide :: Fractional a => a -> a -> a
divide num1 num2 = num1 / num2

File: PrimeraPregunta.py

def suma(x, y):
    return x + y
def resta(x, y):
    return x - y
def multiplicacion(x, y):
    return x * y
def divide(x, y):
    return x / y

def calculadora(x, y, op):
    if op == '+':
        r = suma(x, y)
    if op == '-':
        r = resta(x, y)
    if op == '*':
        r = multiplicacion(x, y)
    if op == '/':
        r = divide(x, y)
    return r;  
print(calculadora(5, 7, '+'))
print(calculadora(5, 7, '*'))
print(calculadora(5, 7, '/'))
print(calculadora(5, 7, '-'))
